src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,155 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        addNode(value, size, tail, null);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Wrong index");
+        }
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> currentNode = getNode(index);
+            currentNode.prev = addNode(value, index, currentNode.prev, currentNode);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexExistence(index);
+        return getNode(index).value;
```

Comment:
This is code repeating. You always check the index before getting node by index. What you should do is move the check logic INSIDE the get method.